packages/react/src/useSecurityKeys.ts

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState } from 'react'
+
+import {
+  ActionErrorState,
+  ActionSuccessState,
+  AddSecurityKeyHandlerResult,
+  addSecurityKeyPromise,
+  ErrorPayload,
+  OTHER_ERROR_CODE
+} from '@nhost/core'
+
+import { useNhostClient } from './useNhostClient'
+
+interface AddSecurityKeyHandler {
+  (
+    /** Optional human-readable name of the security key */
+    nickname?: string
+  ): Promise<AddSecurityKeyHandlerResult>
+}
+interface RemoveSecurityKeyHandlerResult extends ActionErrorState, ActionSuccessState {}
+interface RemoveSecurityKeyHandler {
+  (
+    /** Unique identifier of the security to remove */
+    id: string
+  ): Promise<RemoveSecurityKeyHandlerResult>
+}
+
+export type SecurityKey = { id: string; nickname?: string }
+
+export interface SecurityKeysHookResult extends ActionErrorState, ActionSuccessState {
+  /** Add a security key to the current user with the WebAuthn API */
+  add: AddSecurityKeyHandler
+  /**  List the security keys of the current user */
+  list: SecurityKey[]
+  /** Remove the given security key from the list of allowed keys for the current user */
+  remove: (id: string) => Promise<RemoveSecurityKeyHandlerResult>
```

Comment:
`add` function has `AddSecurityKeyHandler`. I'd suggest to either remove that and inline the `add` function's type in this interface or create one for this too.